### PR TITLE
[REF] Refactor `read_psf_information`

### PR DESCRIPTION
**Pull request by NicolasGensollen**
This PR proposes to refactor the function `clinica.utils.pet.read_psf_information`:

- Add type hints
- Docstring follows Numpydoc conventions
- Refactor code to make it shorter and easier to read
- Add unit tests

### CODE REVIEW

File: clinica/utils/pet.py [hunk 1]
```diff
@@ -18,17 +22,46 @@ class Tracer(str, Enum):
     FMM = "18FFMM"
 
 
-def read_psf_information(pvc_psf_tsv, subject_ids, session_ids, pet_tracer):
+def read_psf_information(
+    pvc_psf_tsv: os.PathLike,
+    subject_ids: ty.List[str],
+    session_ids: ty.List,
```
Comment (ghisvail) [inline clinica/utils/pet.py:28]:
```suggestion
    session_ids: ty.List[str],
```

File: test/unittests/utils/test_pet.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+def test_read_psf_information_errors(tmp_path, psf_df):
+    from clinica.utils.pet import read_psf_information
+
+    with pytest.raises(
+        FileNotFoundError,
+        match="No such file or directory: 'foo.tsv'",
+    ):
+        read_psf_information(
+            "foo.tsv", ["sub-CLNC01", "sub-CLNC01"], ["ses-M00", "ses-M18"], "FDG"
```
Comment (ghisvail) [inline test/unittests/utils/test_pet.py:27]: "Expected type 'PathLike', got 'str' instead"

Not a big deal, but if we don't want type alignment warnings, it should be wrapped in `pathlib.PurePath`.

File: test/unittests/utils/test_pet.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    ):
+        read_psf_information(
+            tmp_path / "wrong_psf_2.tsv",
+            ["sub-CLNC01", "sub-CLNC01"],
+            ["ses-M00", "ses-M18"],
+            "FDG",
+        )
+
+
+def test_read_psf_information(tmp_path, psf_df):
```
Comment (ghisvail) [inline test/unittests/utils/test_pet.py:97]: Missing typing

File: test/unittests/utils/test_pet.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            "session_id": ["ses-M00", "ses-M18"] + ["ses-M00"] * 3,
+            "acq_label": ["FDG", "FDG", "AV45", "FDG", "FDG"],
+            "psf_x": [8, 8, 7, 8, 8],
+            "psf_y": [9, 9, 6, 9, 9],
+            "psf_z": [10, 10, 5, 10, 10],
+        }
+    )
+
+
+def test_read_psf_information_errors(tmp_path, psf_df):
```
Comment (ghisvail) [inline test/unittests/utils/test_pet.py:19]: Missing typing

File: test/unittests/utils/test_pet.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
+import pandas as pd
+import pytest
+
+
+@pytest.fixture
+def psf_df():
```
Comment (ghisvail) [inline test/unittests/utils/test_pet.py:6]:
```suggestion
def psf_df() -> pd.DataFrame:
```

**Review by ghisvail — CHANGES_REQUESTED** [review]
LGTM, with minor corrections

**Comment by NicolasGensollen** [comment]
Thanks for the review @ghisvail !

**Pull request merged**